src/tables.jl

```diff
@@ -1,10 +1,24 @@
 using Tables: Tables
 
-Tables.istable(::Type{<:StructVector}) = true
-Tables.rowaccess(::Type{<:StructVector}) = true
-Tables.columnaccess(::Type{<:StructVector}) = true
+Tables.isrowtable(::Type{<:StructVector}) = true
 
-Tables.rows(s::StructVector) = s
+Tables.columnaccess(::Type{<:StructVector}) = true
 Tables.columns(s::StructVector) = fieldarrays(s)
 
 Tables.schema(s::StructVector) = Tables.Schema(staticschema(eltype(s)))
+
+function Base.append!(s::StructVector, rows)
+    if Tables.isrowtable(rows) && Tables.columnaccess(rows)
+        # Input `rows` is a container of rows _and_ satisfies column
+        # table interface.  Thus, we can add the input column-by-column.
+        table = Tables.columns(rows)
+        nt = foldl(Tables.columnnames(table); init = NamedTuple()) do nt, name
```

Comment:
I'm afraid that this could be a bit heavy on the compiler. Maybe, when iterating pairs `name => Tables.getcolumn(table, name)`, one could just do `append!(getproperty(s, name), Tables.getcolumn(table, name))`, without having to generate all the intermediate named tuples.

It'd be also good to check that this works for  `StructArray{<:Tuple}`, I'm not completely sure how Tables.jl works in the case of integer names (rather than symbol names).

Reply:
integer column names should be fully supported

Reply:
I added a test like this:

https://github.com/JuliaArrays/StructArrays.jl/blob/8f1ce3f18bcfde3ad27784fe59f3f3c7d66d2524/test/runtests.jl#L346-L348

I used `invoke` as I don't know if there is a pre-defined tables with integer column names in Tables.jl.  I tried `Tables.table(([1, 2], [3, 4]))` and `Tables.table([1 2; 3 4])` but they didn't work.

Reply:
`t = Tables.table([1 2; 3 4])` didn't work in the sense that `propertynames(t)` is `[:Column1, :Column2]` rather than `[1, 2]` (even though `getproperty(t, 1)` does work).

Since `Tables.getcolumn` does not have something equivalent to `get(container, key, default)`, there is no exception-safe way to append tables column-by-column, other than checking `Tables.columnnames`.